lib/rubygems/uninstaller.rb

```diff
@@ -42,10 +42,27 @@ class Gem::Uninstaller
   attr_reader :spec
 
   ##
-  # Constructs an uninstaller that will uninstall +gem+
+  # Constructs an uninstaller that will uninstall +gem+ which can be a gem name.
+  # +options+ is a Hash with the following keys:
+  #
+  # :version:: Version requirement for the gem to uninstall. If not specified,
+  #            uses Gem::Requirement.default.
+  # :install_dir:: The directory where the gem was installed. If not specified,
+  #                uses Gem.dir.
+  # :executables:: Force removal of executables. If true, removes executables
+  #                without asking. If false, asks the user. If nil, uses default
+  #                behavior.
+  # :all:: Remove all matching versions without asking the user.
```

Comment:
I'd elaborate a bit more

```suggestion
  # :all:: If more than one version matches the requirement, whether to forcefully remove all matching versions or ask the user to select specific matching versions that should be removed.
```

By the way, I just realized that even if `all` is passed `false`, passing `force` will override it, which seems like a bug?